core/src/main/java/io/github/fourlastor/game/demo/state/unit/Unit.java

```diff
@@ -48,6 +48,11 @@ public void setHp(int hpAmount) {
         refreshHpLabel();
     }
 
+    /**
+     * (sheerst) Note: I would like this to just be Unit.image.getPosition() etc.
+     * Even more ideal would be something like the auto getters/setters in C#.
+     * ie Unit.image.position = something  <- calls setter
+     */
```

Comment:
that would be Kotlin :slightly_smiling_face: 

```kotlin
fun Actor.getPosition(): Vector2 = vec2(x, y)
```
Using inheritance for this is a no go, it would mean two possible courses of action:
1. Extending every actor type we use  to be able to do this, and keep doing it for each new actor type, and add this method (otherwise we have actors which have a different api that we can rely upon, while they should maintain the same level of abstraction and capabilities)
2. Having each of our actors use a base class, which strongly limits our options in terms of what they can do, and tends to end up being a god class full of unrelated methods

I don't think either of them are worth the convenience of creating a `Vector2`